### PR TITLE
refactor(query): streaming load use planner v2.

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

part of https://github.com/datafuselabs/databend/issues/7732

```StreamingWithFormat(format, pipe)``` will be replaced with ```StreamingWithFormat(format, input_context)```
after https://github.com/datafuselabs/databend/pull/7613 is merges

### CODE REVIEW

**Review by zhang2014 — APPROVED** [review]

File: src/query/service/src/sql/planner/plans/insert.rs [hunk 1]
```diff
@@ -23,7 +24,8 @@ pub enum InsertInputSource {
     #[serde(skip)]
     SelectPlan(Box<Plan>),
     // From outside streaming source
-    StreamingWithFormat(String),
+    #[serde(skip)]
```
Comment (Xuanwo) [inline src/query/service/src/sql/planner/plans/insert.rs:27]: Can we remove `serde::Serialize, serde::Deserialize` for `InsertInputSource`? Do we really need this?

Comment (youngsofun) [reply src/query/service/src/sql/planner/plans/insert.rs:27]: no idea， maybe for distributed execute？ cc @zhang2014 @sundy-li

File: src/query/service/src/servers/http/v1/load.rs [hunk 2]
```diff
@@ -55,32 +51,15 @@ pub struct LoadResponse {
...
     async move {
-        let interpreter = InterpreterFactory::get(context.clone(), node).await?;
+        let interpreter = InterpreterFactoryV2::get(context.clone(), &plan).await?;
 
         if let Err(cause) = interpreter.start().await {
             error!("interpreter.start error: {:?}", cause);
         }
 
-        // TODO(Winter): very hack code. need remove it.
-        interpreter.set_source_pipe_builder(Option::from(source_builder))?;
```
Comment (youngsofun) [inline src/query/service/src/servers/http/v1/load.rs:82]: @zhang2014 still used by async insert

File: src/query/service/src/sql/planner/plans/insert.rs (hunk 1 above)
Comment (youngsofun) [reply src/query/service/src/sql/planner/plans/insert.rs:27]: still ok here because StreamingWithFormat can only run in one node.

Comment (youngsofun) [reply src/query/service/src/sql/planner/plans/insert.rs:27]:
```
#[derive(serde::Serialize, serde::Deserialize, Clone)]
pub struct Insert {
   ...
    pub source: InsertInputSource,
}

#[derive(Clone, Debug)]
pub enum Plan {
  ... 
  Insert(Box<Insert>),
}
```
but enum Plan  not impl Serialize

**Review by sundy-li — APPROVED** [review]

**Pull request merged**